src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1294,7 +1294,7 @@ void ImportStoreIndirect(TypeDesc type)
 
             CheckIsByRef(address);
             if (!value.IsNullReference)
-                CheckIsAssignable(type, address.Type);
+                CheckIsAssignable(StackValue.CreateFromType(type), StackValue.CreateFromType(address.Type));
```

Comment:
Nit: I know that perf is not a top concern for ILVerify, but it may be nice to avoid calling `StackValue.CreateFromType(type)` twice.